### PR TITLE
[Docs] Add some notes about native file dialog limitations.

### PR DESCRIPTION
**Pull request by bruvzg**

### CODE REVIEW

File: doc/classes/FileDialog.xml [hunk 1]
```diff
@@ -80,12 +80,14 @@
 			<description>
 				Returns the vertical box container of the dialog, custom controls can be added to it.
 				[b]Warning:[/b] This is a required internal node, removing and freeing it may cause a crash. If you wish to hide it or any of its children, use their [member CanvasItem.visible] property.
+				[b]Note:[/b] Changes to this node are ignored by native file dialogs, use [method add_option] to add custom element to the dialog instead.
```
Comment (Mickeon) [inline doc/classes/FileDialog.xml:83]:
```suggestion
				[b]Note:[/b] Changes to this node are ignored by native file dialogs, use [method add_option] to add custom elements to the dialog instead.
```

File: doc/classes/FileDialog.xml [hunk 2]
```diff
@@ -80,12 +80,14 @@
...
 			</description>
 		</method>
 		<method name="invalidate">
 			<return type="void" />
 			<description>
 				Invalidate and update the current dialog content list.
+				[b]Note:[/b] This method is ignored by native file dialogs.
```
Comment (Mickeon) [inline doc/classes/FileDialog.xml:90]: You may as well say, perhaps...?
```suggestion
				[b]Note:[/b] This method does nothing on native file dialogs.
```

File: doc/classes/FileDialog.xml [hunk 3]
```diff
@@ -116,10 +118,11 @@
 	<members>
 		<member name="access" type="int" setter="set_access" getter="get_access" enum="FileDialog.Access" default="0">
 			The file system access scope. See [enum Access] constants.
-			[b]Warning:[/b] Currently, in sandboxed environments such as Web builds or sandboxed macOS apps, FileDialog cannot access the host file system. See [url=https://github.com/godotengine/godot-proposals/issues/1123]godot-proposals#1123[/url].
+			[b]Warning:[/b] In Web builds, FileDialog cannot access the host file system. In sandboxed Linux and macOS environments, [member use_native_dialog] is automatically used to allow limited access to host file system.
 		</member>
 		<member name="current_dir" type="String" setter="set_current_dir" getter="get_current_dir">
 			The current working directory of the file dialog.
+			[b]Note:[/b] This property might be ignored by native file dialogs.
```
Comment (Mickeon) [inline doc/classes/FileDialog.xml:125]: This note is way too broad and, from the reader's point of view, it's meaningless. If anything, it implies the property is unreliable and you shouldn't use it.

Comment (bruvzg) [reply doc/classes/FileDialog.xml:125]: > it implies the property is unreliable and you shouldn't use it.

It is unreliable.

For the reference, description for the underlying native implementation docs:

> Suggested folder from which the files should be opened.
> The portal implementation is free to ignore this option.

Comment (Mickeon) [reply doc/classes/FileDialog.xml:125]: It's worth briefly explaining *why* that's the case, then, instead of leaving the reader on its own thoughts.

**Review by akien-mga — APPROVED** [review]
Looks good to me overall. Agree with Mickeon about the note that could be made clearer.

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**